henriqueApp/hooks/useLogin.tsx

```diff
@@ -0,0 +1,59 @@
+import { useState } from "react";
+import { Alert } from "react-native";
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import {useEmailValidation} from "./useEmailValidation";
+import {usePasswordValidation} from "./usePasswordValidation";
+
+const API_URL = 'https://template-onboarding-node-sjz6wnaoia-uc.a.run.app/authenticate';
+
+export function useLogin() {
+  const { email, setEmail, emailError, validateEmailField } = useEmailValidation();
+  const { password, setPassword, passwordError, validatePasswordField } = usePasswordValidation();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [serverError, setServerError] = useState<string | null>(null);
+
+  const handleSubmit = async () => {
+    const isEmailValid = validateEmailField();
+    const isPasswordValid = validatePasswordField();
+
+    if (isEmailValid && isPasswordValid) {
+      setIsLoading(true);
+
+      try {
+        const response = await fetch(API_URL, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({ email, password }),
+        });
+
+        const data = await response.json();
+
+        if (response.ok) {
+          await AsyncStorage.setItem('authToken', data.data.token);
+        } else {
+          setServerError(data.errors[0].message || "Erro desconhecido.");
```

Comment:
Por você ter essa lógica de fallback `|| "Erro desconhecido"`, acho que seria o caso então de evitar que ele jogue erro aqui caso não venha a estrutura esperada:

```suggestion
          setServerError(data.errors?.[0]?.message ?? "Erro desconhecido.");
```

- Usando "optional chaining" para que ele não jogue erro tentando acessar um campo de algo undefined
- Usando o "nullish coalescing" `??` pq ele é mais adequado que o `||` pra esses cenários